Sources/Logging/Logging.swift

```diff
@@ -707,16 +721,19 @@ extension Logger {
     ///    over `..., metadata: ["colors": .array([.string("\(user.topColor)"), .string("\(user.secondColor)")])`
     ///  - prefer `logger.info("nested info", metadata: ["nested": ["fave-numbers": ["\(1)", "\(2)", "\(3)"], "foo": "bar"]])`
     ///    over `..., metadata: ["nested": .dictionary(["fave-numbers": ...])])`
-    public enum MetadataValue {
+    public enum MetadataValue: _SwiftLogSendable {
         /// A metadata value which is a `String`.
         ///
         /// Because `MetadataValue` implements `ExpressibleByStringInterpolation`, and `ExpressibleByStringLiteral`,
         /// you don't need to type `.string(someType.description)` you can use the string interpolation `"\(someType)"`.
         case string(String)
 
         /// A metadata value which is some `CustomStringConvertible`.
+        #if compiler(>=5.6)
+        case stringConvertible(CustomStringConvertible & Sendable)
```

Comment:
Why does the value need to be Sendable here?

Reply:
Because otherwise `MetadataValue` cannot conform to `Sendable`, which in term means `Logger.Metadata` cannot conform to `Sendable`, which in the end means `Logger` can't conform to `Sendable`.

Reply:
> Because otherwise MetadataValue cannot conform to Sendable, which in term means Logger.Metadata cannot conform to Sendable, which in the end means Logger can't conform to Sendable.

Yeah, I get that. However we could also transform `CustomStringConvertible` to a string directly on the call-site. This would allow us to not require `Sendable`. However a class that might change its string representation at a later point, would always be represented by its original value, if it is added to the metadata. 

> This wouldn't work either, because this would be a new protocol. Currently, it only requires `CustomStringConvertible`.

Very good point. Going directly to string might potentially allow us to work around the semver major here.

Reply:
@DougGregor any advice here to avoid the need in a major API breakage?

Reply:
> Yeah, I get that. However we could also transform CustomStringConvertible to a string directly on the call-site. 

That's not an option; the entire purpose of this case is to delay rendering the string; @ffried mentioned this already I just noticed.

Maybe a controversial take but... the `& Sendable` will not cause errors (!), until Swift 6... so, I'm not sure we need to consider it breaking -- rather, this gives people the right `warning` and they have time to fix the issue before it turns into an error.

Reply:
@ktoso are you saying that adding the Sendable protocol requirement will only yield warnings until Swift 6?

Reply:
That's right, I just confirmed with Doug as well. If you were seeing an error in beta-1 (maybe?), then Doug says this could have been a bug and is still just a warning nowadays and will continue to be.

Reply:
in that case I am also comfortable with this soft break. 

@ffried @fabianfett @weissi @Lukasa wdyt?

Reply:
Same here. A warning is fine. 👍
It's the same as when for example a new deprecation is added. There would also be a warning then.

Reply:
I'm ok with a warning here.